### ISSUE TITLE
Deselecting the last mail item does not fire any events to notify the add-in in Outlook for Desktop

### ISSUE THREAD

**Issue by VasilyIvanov** [OfficeDev/office-js#3846]
We are developing an Outlook add-in that should be able to work with multiple selected emails as documented [here](https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/item-multi-select).
While testing in Outlook Desktop we noticed a problem that appears when we deselect previously selected email items. In fact, our add-in is not notified when the user deselects the last email.
According to the documentation, there are 2 events which are fired when the user changes selection in Outlook, in particular, `Office.EventType.ItemChanged` and `Office.EventType.SelectedItemsChanged`. However, add-ins get no information when the user deselects the last email item.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC desktop
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: Microsoft® Outlook® for Microsoft 365 MSO (Version 2310 Build 16.0.16924.20054) 64-bit
* Operating System: Windows 11
* Browser (if using Office on the web): N/A
 
## Expected behavior
The engine should fire an event that could notify an add-in that no email items are currntly selected.

## Current behavior
An add-in has no chance to get notified that the user has deselected all email items, that is, zero emails are currently selected.

## Steps to reproduce

1. Start an add-in that subscribes to the `Office.EventType.SelectedItemsChanged` event in Outlook Desktop.
2. Select 2 emails. You will get an event showing 2 emails selected.
3. Deselect one email (this could be done by holding the Ctrl key and clicking the email). You will get an event showing 1 email selected.
4. Deselect the last item. You will **not** get an event showing that zero emails are selected.

## Link to live example(s)

1. [This test add-in](https://github.com/VasilyIvanov/TestAddIn) could be used to reproduce but any other add-in that subscribes to the `Office.EventType.SelectedItemsChanged` event will work for you.

# Provide additional details

No extra explanation should be required.

## Context

There is an obvious inconsistency between how the Outlook app displays the selected emails and how an add-in would get information about selected items. I.e. Outlook displays no selected emails but doesn't notify the add-in that all selected items have been deselected.

## Useful logs

No logs are provided because the nature of the issue is that the engine does **not** do what is expected.

**Comment by zhngx1** [comment]
Thanks for reporting this issue regarding _Deselecting the last mail item does not fire any events to notify the add-in in Outlook for Desktop_. It has been put on our backlog. We unfortunately have no timelines to share at this point.

 

Internal tracking id: Office: 8565220

**Comment by zhngx1** [comment]
One thing to clarify, does the Office.EventType.ItemChanged event fire when you deselect the last email.

**Comment by VasilyIvanov** [comment]
Yes, it does. It seems that we are able to create a workaround if we subscribe to both (`ItemChanged `and `SelectedItemsChanged`) events.
However, we believe that the `SelectedItemsChanged` event still **should be fired** because the selected items have been changed. By design, it's supposed to reflect changes in email selection.

**Comment by zhngx1** [comment]
Thanks for reporting this issue regarding _Deselecting the last mail item does not fire any events to notify the add-in in Outlook for Desktop_. It has been put on our backlog. We unfortunately have no timelines to share at this point 

 

Internal tracking id: Office: 8565220

**Comment by zhngx1** [comment]
This issue was fixed in build 16.0.17726.20000

**Comment by mrebuffet** [comment]
This is still an issue in the following version. 
```
osfweb version: 16.0.17824.15000
office-js-api version: 20250115.2
```

Can you please confirm if the fix has been deployed yet?
